src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -3732,48 +3742,139 @@ public enum DarknessAblaze implements LogicCardInfo {
       return itemCard (this) {
         text "Search your deck for a Pokémon with the same name as a Pokémon in your discard pile, and add it to your hand. Then, shuffle your deck."
         onPlay {
+          def pokeNamesInDiscard = []
+          my.discard.filterByType(POKEMON).each{
+            pokeNamesInDiscard.add(it.name)
+          }
+          my.deck.search("Pokémon with same name as a Pokémon in your discard pile. Possible options: $pokeNamesInDiscard", {it.cardTypes.is(POKEMON) && pokeNamesInDiscard.contains(it.name)}).moveTo(my.hand)
         }
         playRequirement{
+          my.deck : "You have no cards in your deck"
+          my.discard.filterByType(POKEMON) : "You have no Pokémon in your discard pile"
         }
       };
       case PIERS_173:
       return supporter (this) {
-        text "Search your deck for 1 [D] Pokémon and any 1 Energy, reveal them, and put them into your hand. Then, shuffle your deck."
-        onPlay {
-        }
-        playRequirement{
-        }
+        text "Search your deck for a [D] Pokémon and an Energy card, reveal them, and put them into your hand. Then, shuffle your deck."
+          onPlay {
+            my.deck.search(count: 1, "Search your deck for a [D] Pokémon",pokemonTypeFilter(D)).moveTo(my.hand)
+            my.deck.search("Search your deck for an Energy card",cardTypeFilter(ENERGY)).moveTo(my.hand)
+            shuffleDeck()
+          }
+          playRequirement{
+            assert my.deck.notEmpty
+          }
       };
       case POKEMON_BREEDER_S_NURTURING_174:
       return supporter (this) {
         text "Choose up to 2 of your Pokémon in play. Search your deck for a card that evolves from each of those Pokémon and put it on top of the Pokémon to evolve it. Then, shuffle your deck. (You can’t use this card during your first turn or on a Pokémon that just came into play.)"
         onPlay {
+          def pcsList = my.all.findAll {it.turnCount < bg.turnCount}
+          if (pcsList.size() > 1) {
+            pcsList = pcsList.select(max:2, "Choose up to 2 of your Pokémon in play that can make a regular evolve during this turn.")
```

Comment:
If this ends up using the CardList select, I think you need to set the minimum as well since you can't play a Trainer without any effect. If this ends up using a PcsList instead you might end up having to cast it to a CardList since the PcsList doesn't have min/max/count parameters from what I see.

Reply:
@flagrama  is correct. `pcsList.select` returns a single `pcs` not a list.

Reply:
Resolved with commit

Reply:
I see. Will correct this so it chooses one by one then.